src/NServiceBus.TransactionalSession/TransactionalSessionDelayControlMessageBehavior.cs

```diff
@@ -31,23 +31,25 @@ public async Task Invoke(IIncomingPhysicalMessageContext context,
         var messageId = context.MessageId;
         if (remainingCommitDuration <= TimeSpan.Zero)
         {
-            if (Log.IsInfoEnabled)
+            if (Log.IsWarnEnabled)
             {
-                Log.Info(
-                    $"Consuming transaction commit control message for the message ID '{messageId}' because the maximum commit duration has elapsed.");
+                Log.WarnFormat(
+                    "Consuming transaction commit control message for message ID '{0}' because the maximum commit duration has elapsed. If this occurs repeatedly, consider increasing the {1} setting on the session.", messageId, nameof(OpenSessionOptions.MaximumCommitDuration));
             }
 
             return;
         }
 
-        if (Log.IsDebugEnabled)
-        {
-            Log.Debug($"Delaying transaction commit control message for the message ID '{messageId}'");
-        }
-
         var newCommitDelay = commitDelayIncrement.Add(commitDelayIncrement);
         commitDelayIncrement = newCommitDelay > remainingCommitDuration ? remainingCommitDuration : newCommitDelay;
 
+        if (Log.IsInfoEnabled)
+        {
+            Log.InfoFormat(
+                "Delaying transaction commit control message for message ID '{0}' by {1} seconds. If this occurs repeatedly, consider increasing the {2} setting on the session.",
```

Comment:
Nitpick: Since we are not delaying the initial send of the control message, this could also indicate that the initial commit of the outbox record is slow to commit for some reason or that the transport used is really fast.

Is there some way we could detect that this is the first time we extend the delay and tailor the message based on that?

Reply:
Another option would be to say "if this occurs repeatedly **for the same message id**..." to highlight that extending the delay only makes sense when there are repeated delays

Reply:
@andreasohlund Yes, this is something we discussed. It would only be the one time, and the discussion that we had was:
- What do we want to optimise for? The user's whos code is the fastest? Or the slowest?

We don't want to slow anyone down that would be as fast as that message arriving to the queue and being processed, and therefore decided not to change anything there.

Reply:
> Is there some way we could detect that this is the first time we extend the delay and tailor the message based on that?

The message is shown for the first time after the first control messages if processed and the outbox record is not there. Am I missing something here?

Reply:
This is shown every time the message is delayed